### PR TITLE
Abstraction contrasts on utils_contrasts

### PR DESCRIPTION
**Pull request by ferponcem**
Now for real, sorry for the messy steps!
This PR contains some changes made to `utils_contrasts` in order to add the contrasts we discussed [here](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Abstraction.csv) for **Abstraction** task. 
@bthirion could you have a look?

### CODE REVIEW

File: ibc_public/utils_contrasts.py [hunk 1]
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
     _append_effects_interest_contrast(design_matrix_columns, contrasts)
     return contrasts
 
+def abstraction(design_matrix_columns):
+    """ Contrasts for Abstraction """
+    localizer_ = ['localizer_faces', 'localizer_humanbody',
```
Comment (bthirion) [inline ibc_public/utils_contrasts.py:285]: Localizer should be another block I think

File: ibc_public/utils_contrasts.py [hunk 2]
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
...
+    flora_ =['flora_geometry', 'flora_edge', 'flora_photo']
+    objects_ = ['objects_geometry', 'objects_edge', 'objects_photo']
+    places_ = ['places_geometry', 'places_edge', 'places_photo']
+    others_ = ['response', 'localizer_faces-others',
+               'localizer_humanbody-others','localizer_words-others',
+               'localizer_nonsensewords-others',
+               'localizer_numbers-others',
+               'localizer_places-others', 'localizer_objects-others',
+               'localizer_checkerboards-others','humanbody-other',
+               'animals-other','faces-other',
```
Comment (bthirion) [inline ibc_public/utils_contrasts.py:302]: be consistent on naming: 'other' vs 'others'

File: ibc_public/utils_contrasts.py [hunk 3]
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
...
+        localizer
+    contrasts['localizer_checkerboards-others'] = 7 *\
+        con['localizer_checkerboards'] - localizer
+    contrasts['humanbody-other'] = 5 * humanbody - allstim
+    contrasts['animals-other'] = 5 * animals - allstim
+    contrasts['faces-other'] = 5 * faces- allstim
+    contrasts['flora-other'] = 5 * flora - allstim
+    contrasts['places-other'] = 5 * places - allstim
+    contrasts['objects-other'] = 5 * objects - allstim
+    contrasts['geometry-others'] = 2 * geometry - allstim
```
Comment (bthirion) [inline ibc_public/utils_contrasts.py:373]:
```suggestion
    contrasts['geometry-others'] = 3 * geometry - allstim
``` as we want the contrasts to sum to 0 ?

Comment (bthirion) [reply ibc_public/utils_contrasts.py:373]: Same thing for the following contrasts.

**Comment by ferponcem** [comment]
Thanks for the feedback!
Some changes I made:
* Separated the localizer as a different block, now we have both `Abstraction` and `AbstractionLocalizer`. See changes in the script. Along to that I made the changes on the [conditions file](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_conditions.tsv#L389), on the contrast [sheet](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Abstraction.csv) and added a [new one](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/AbstractionLocalizer.csv) for the localizer.
* Modified the weights for the contrasts as we want them to sum up to zero,hopefully like this is alright for this format!
* For the sake of consistency, I am using `other` for the contrasts names

Let me know what you think @bthirion !

File: ibc_public/utils_contrasts.py [hunk 4]
```diff
@@ -279,9 +283,158 @@ def multi_modal(design_matrix_columns):
...
+    con = _elementary_contrasts(design_matrix_columns)
+    contrasts = dict([(name, con[name]) for name in localizer_])
+    localizer = np.sum([con[x] for x in localizer_], 0)
+    contrasts['localizer_faces-other'] = 8 * con['localizer_faces'] -\
+        localizer
+    contrasts['localizer_humanbody-other'] = 8 *\
+        con['localizer_humanbody'] - localizer
+    contrasts['localizer_words-other'] = 8 * con['localizer_words'] -\
+        localizer
+    contrasts['localizer_nonsensewords-other'] = 7 *\
```
Comment (bthirion) [inline ibc_public/utils_contrasts.py:311]: Why 7 ?

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

File: ibc_public/utils_contrasts.py (hunk 4 above)
Comment (ferponcem) [reply ibc_public/utils_contrasts.py]: Fixed! Thanks!

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**